credential/cargo-credential/src/stdio.rs

```diff
@@ -0,0 +1,163 @@
+use std::{fs::File, io::Error};
+
+/// Reset stdin and stdout to the attached console / tty for the duration of the closure.
+/// If no console is available, stdin and stdout will be redirected to null.
+pub fn stdin_stdout_to_console<F, T>(f: F) -> Result<T, Error>
+where
+    F: FnOnce() -> T,
+{
+    let open_write = |f| std::fs::OpenOptions::new().write(true).open(f);
+
+    let mut stdin = File::open(imp::IN_DEVICE).or_else(|_| File::open(imp::NULL_DEVICE))?;
```

Comment:
If both `/dev/tty` and `/dev/null` don't exist, you get something like [this](https://users.rust-lang.org/t/rust-broken-on-second-invocation-in-alpine-linux-chroot/112113?u=correabuscar) whereby `/dev/null` is a file now and some previous output will now be interpreted and fail compilation (or worse).